src/Tools/VMC/EventLibraryInterface.cxx

```diff
@@ -0,0 +1,261 @@
+//____________________________________________________________________________
+/*
+ Copyright (c) 2003-2020, The GENIE Collaboration
+ For the full text of the license visit http://copyright.genie-mc.org
+
+
+*/
+//____________________________________________________________________________
+
+#include "Framework/Messenger/Messenger.h"
+#include "Framework/Numerical/RandomGen.h"
+#include "Framework/GHEP/GHepRecord.h"
+#include "Framework/GHEP/GHepParticle.h"
+#include "Framework/ParticleData/PDGUtils.h"
+#include "Framework/ParticleData/PDGLibrary.h"
+#include "Framework/Interaction/Interaction.h"
+#include "Tools/VMC/EventLibraryInterface.h"
+#include "Tools/VMC/RecordList.h"
+#include "Tools/VMC/Utils.h"
+
+#include "TFile.h"
+
+using namespace genie;
+using namespace genie::vmc;
+
+//____________________________________________________________________________
+EventLibraryInterface::EventLibraryInterface() :
+  EventRecordVisitorI("genie::vmc::EventLibraryInterface"),
+  fRecordFile(0)
+{
+
+}
+//____________________________________________________________________________
+EventLibraryInterface::EventLibraryInterface(string config) :
+  EventRecordVisitorI("genie::vmc::EventLibraryInterface", config),
+  fRecordFile(0)
+{
+
+}
+//____________________________________________________________________________
+EventLibraryInterface::~EventLibraryInterface()
+{
+  for(auto it: fRecords) delete it.second;
+  delete fRecordFile;
+}
+//____________________________________________________________________________
+void EventLibraryInterface::ProcessEventRecord(GHepRecord * event) const
+{
+// Get event summary constructed by GENIE
+//
+  Interaction* interaction = event->Summary();
+  const InitialState & init_state = interaction->InitState();
+
+  const Record* rec = GetRecord(interaction);
+  if(!rec) return; // Reason has already been printed
+
+  std::unique_ptr<TLorentzVector> probe_p4(init_state.GetProbeP4(kRfLab));
```

Comment:
This is not ok if we want to release it in v 3.2, otherwise it will be ok

Reply:
I had no idea genie was compatible with such ancient standards. I take it the plan is to drop that requirement soon?

Reply:
How do I set my built up to use C++98? There look to be a lot of places and I'm likely to miss one.

Alternately -- is it possible to have this component optional, and require a higher compiler version if it's enabled? I don't think any potential users are on anything like so old a standard.